From 6cb6acb44a1cff4136125faf763e67525ea696b1 Mon Sep 17 00:00:00 2001
From: Hassan Eslami <heslami@fb.com>
Date: Wed, 15 Jul 2015 12:48:22 -0700
Subject: [PATCH] [GIRAPH-1020] TaskInfo equality condition bug fix

Summary: Currently equality is checked based on the raw host-name of an object with lower-case host-name of another object. This is not the right semantic and can cause subsequent bugs in partition assignment and migration.

Test Plan: mvn clean verify

Reviewers: maja.kabiljo, sergey.edunov, dionysis.logothetis, avery.ching

Reviewed By: dionysis.logothetis

Differential Revision: https://reviews.facebook.net/D42273
---
 CHANGELOG                                                       | 2 ++
 giraph-core/src/main/java/org/apache/giraph/graph/TaskInfo.java | 2 +-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/CHANGELOG b/CHANGELOG
index 79f95b00d..e72881037 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -2,6 +2,8 @@ Giraph Change Log
 
 Release 1.2.0 - unreleased
 =======
+  GIRAPH-1020: TaskInfo equality condition. (heslami via aching)
+
   GIRAPH-1019: Optimizing and debugging vertex mutation. (heslami via aching)
 
   GIRAPH-1018: Improving PartitionStore API to better match its expected behaviour
diff --git a/giraph-core/src/main/java/org/apache/giraph/graph/TaskInfo.java b/giraph-core/src/main/java/org/apache/giraph/graph/TaskInfo.java
index d593204e2..ab3ac649d 100644
--- a/giraph-core/src/main/java/org/apache/giraph/graph/TaskInfo.java
+++ b/giraph-core/src/main/java/org/apache/giraph/graph/TaskInfo.java
@@ -110,7 +110,7 @@ public String getHostnameId() {
   public boolean equals(Object other) {
     if (other instanceof TaskInfo) {
       TaskInfo taskInfo = (TaskInfo) other;
-      if (hostname.equals(taskInfo.getHostname()) &&
+      if (getHostname().equals(taskInfo.getHostname()) &&
           (getTaskId() == taskInfo.getTaskId()) &&
           (port == taskInfo.getPort() &&
           (taskId == taskInfo.getTaskId()))) {
